Automattic/jetpack
# Premium Content: Check for required plan on jetpack sites

Fixes Automattic/view-design#158

#### Changes proposed in this Pull Request:

* Adds a condition to the `access_checks` for Jetpack sites, in order to check whether the site has the required plan upgrade necessary to render the Premium Content block. This is done so that the Upgrade Nudge will be rendered correctly on the frontend(instead of defaulting to the Stripe Nudge) when the block isn't configured correctly.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






Since this bug is masked behind Automattic/view-design#160, (block doesn't appear at all on free Jetpack sites), you need to stub out the plan checks in order to test. Remove the `plan_check` from the Premium Content block registration [here](https://github.com/Automattic/jetpack/blob/b7ae39a4a6304f3dafd3622fd1c08243ab23bd4f/extensions/blocks/premium-content/premium-content.php#L35)

(Setup instructions taken from #17907)
* Please use https://github.com/Automattic/a8c-wp-env to test as it will make your life much easier.
* Start up a site with npm run start, and DO NOT include Editing Toolkit (just use Gutenberg, and Jetpack).
* Check out this branch and activate jurassic.tube with npm run tube:init
* ** Verify that your site does NOT have a paid Jetpack plan while testing this**
* Create a new post and insert a Premium Content block
* Preview the post on the frontend and verify that you see the Logged Out view with the Upgrade Nudge
* Add a "Security Daily" Jetpack plan to your site
* Return to the post and open the preview again. You should now see the Stripe Nudge

<img width="635" alt="Screen Shot 2020-12-09 at 5 06 56 PM" src="https://user-images.githubusercontent.com/63313398/102256635-786d3d00-3ec0-11eb-9960-2a5c1e4f7ad2.png">
<img width="615" alt="Screen Shot 2020-12-09 at 5 09 56 PM" src="https://user-images.githubusercontent.com/63313398/102256649-7d31f100-3ec0-11eb-96da-be50dc23a94f.png">
 

#### Proposed changelog entry for your changes:


*
